leases/utils.py

```diff
@@ -159,6 +159,34 @@ def calculate_winter_storage_lease_end_date() -> date:
     return calculate_winter_season_end_date()
 
 
+def calculate_lease_termination_date(
+    lease: Union[BerthLease, WinterStorageLease], end_date: date = None
+) -> date:
+    from .models import BerthLease
+
+    # The highest priority goes to the parameter, if the end_date is specified,
+    # it will be set, no matter the checks. It's assumed to be a valid date.
+    if end_date:
+        return end_date
+
+    # If the lease hasn't started, the lease will end on the same date as it starts.
+    elif not lease.has_started:
+        return lease.start_date
+
+    # If the lease is terminated during the season, the end date will be set to the
+    # current date.
+    elif lease.is_ongoing:
+        return today().date()
+
+    # If the lease is terminated after its season has ended, the end date will be set
+    # to the date when the season ends.
+    elif lease.has_ended:
+        if isinstance(lease, BerthLease):
+            return calculate_season_end_date(lease.start_date)
+        else:  # WinterStorageLease
+            return calculate_winter_season_end_date(lease.start_date)
+
+
```

Comment:
What if none of the `if`s matches, should this then return `None` as it seems to do now? Then at least the return type hint is slightly wrong.

Reply:
Good point, this last case returns the default end date for the lease, so it should return it in the last case.